{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quick mapping of reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "BSMAP v2.74\r\n",
      "Usage:\tbsmap [options]\r\n",
      "       -a  <str>   query a file, FASTA/FASTQ/BAM format\r\n",
      "       -d  <str>   reference sequences file, FASTA format\r\n",
      "       -o  <str>   output alignment file, BSP/SAM/BAM format, if omitted, the output will be written to STDOUT in SAM format.\r\n",
      "\r\n",
      "  Options for alignment:\r\n",
      "       -s  <int>   seed size, default=16(WGBS mode), 12(RRBS mode). min=8, max=16.\r\n",
      "       -v  <float> if this value is between 0 and 1, it's interpreted as the mismatch rate w.r.t to the read length.\r\n",
      "                   otherwise it's interpreted as the maximum number of mismatches allowed on a read, <=15.\r\n",
      "                   example: -v 5 (max #mismatches = 5), -v 0.1 (max #mismatches = read_length * 10%)\r\n",
      "                   default=0.08.\r\n",
      "       -g  <int>   gap size, BSMAP only allow 1 continuous gap (insert or deletion) with up to 3 nucleotides\r\n",
      "                   default=0\r\n",
      "       -w  <int>   maximum number of equal best hits to count, <=1000\r\n",
      "       -3          using 3-nucleotide mapping approach\r\n",
      "       -B  <int>   start from the Nth read or read pair, default: 1\r\n",
      "       -E  <int>   end at the Nth read or read pair, default: 4,294,967,295\r\n",
      "       -I  <int>   index interval, default=4\r\n",
      "       -k  <float> set the cut-off ratio for over-represented kmers, default=5e-07\r\n",
      "                   example: -k 1e-6 means the top 0.0001% over-represented kmer will be skipped in alignment\r\n",
      "       -p  <int>   number of processors to use, default=8\r\n",
      "       -D  <str>   activating RRBS mapping mode and set restriction enzyme digestion sites. \r\n",
      "                   digestion position marked by '-', example: -D C-CGG for MspI digestion.\r\n",
      "                   default: none (whole genome shotgun bisulfite mapping mode)\r\n",
      "       -S  <int>   seed for random number generation used in selecting multiple hits\r\n",
      "                   other seed values generate pseudo random number based on read index number, to allow reproducible mapping results. \r\n",
      "                   default=0. (get seed from system clock, mapping results not resproducible.)\r\n",
      "       -n  [0,1]   set mapping strand information. default: 0\r\n",
      "                   -n 0: only map to 2 forward strands, i.e. BSW(++) and BSC(-+), \r\n",
      "                   for PE sequencing, map read#1 to ++ and -+, read#2 to +- and --.\r\n",
      "                   -n 1: map SE or PE reads to all 4 strands, i.e. ++, +-, -+, -- \r\n",
      "       -M  <str>   set alignment information for the additional nucleotide transition. \r\n",
      "                   <str> is in the form of two different nucleotides N1N2, \r\n",
      "                   indicating N1 in the reads could be mapped to N2 in the reference sequences.\r\n",
      "                   default: -M TC, corresponds to C=>U(T) transition in bisulfite conversion. \r\n",
      "                   example: -M GA could be used to detect A=>I(G) transition in RNA editing. \r\n",
      "\r\n",
      "  Options for trimming:\r\n",
      "       -q  <int>   quality threshold in trimming, 0-40, default=0 (no trim)\r\n",
      "       -z  <int>   base quality, default=33 [Illumina is using 64, Sanger Institute is using 33]\r\n",
      "       -f  <int>   filter low-quality reads containing >n Ns, default=5\r\n",
      "       -A  <str>   3-end adapter sequence, default: none (no trim)\r\n",
      "       -L  <int>   map the first N nucleotides of the read, default:144 (map the whole read).\r\n",
      "\r\n",
      "  Options for reporting:\r\n",
      "       -r  [0,1]   how to report repeat hits, 0=none(unique hit/pair only); 1=random one, default:1.\r\n",
      "       -R          print corresponding reference sequences in SAM output, default=off\r\n",
      "       -u          report unmapped reads, default=off\r\n",
      "       -H          do not print header information in SAM format output\r\n",
      "\r\n",
      "  Options for pair-end alignment:\r\n",
      "       -b  <str>   query b file\r\n",
      "       -m  <int>   minimal insert size allowed, default=28\r\n",
      "       -x  <int>   maximal insert size allowed, default=500\r\n",
      "\r\n",
      "       -h          help\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/BSMAP/bsmap-2.74/bsmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 12.8M  100 12.8M    0     0  7965k      0  0:00:01  0:00:01 --:--:-- 7969k\n"
     ]
    }
   ],
   "source": [
    "!curl http://eagle.fish.washington.edu/cnidarian/OlyO_Pat_PacBio_10k_contigs.fa \\\n",
    "-o /Volumes/Histidine/hectocotylus/OlyO_Pat_PacBio_10k_contigs.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "BSMAP v2.74\n",
      "Start at:  Fri Jan 22 12:23:45 2016\n",
      "\n",
      "Input reference file: /Volumes/Histidine/hectocotylus/OlyO_Pat_PacBio_10k_contigs.fa \t(format: FASTA)\n",
      "Load in 553 db seqs, total size 13282692 bp. 0 secs passed\n",
      "total_kmers: 43046721\n",
      "Create seed table. 2 secs passed\n",
      "max number of mismatches: read_length * 8% \tmax gap size: 0\n",
      "kmer cut-off ratio: 5e-07\n",
      "max multi-hits: 100\tmax Ns: 5\tseed size: 16\tindex interval: 4\n",
      "quality cutoff: 0\tbase quality char: '!'\n",
      "min fragment size:28\tmax fragemt size:500\n",
      "start from read #1\tend at read #4294967295\n",
      "additional alignment: T in reads => C in reference\n",
      "mapping strand: ++,-+\n",
      "Single-end alignment(8 threads)\n",
      "Input read file: /Volumes/web-1/nightingales/O_lurida/1_ATCACG_L001_R1_001.fastq.gz \t(format: gzipped FASTQ)\n",
      "Output file: /Volumes/Histidine/hectocotylus/1_ATCACG_BSMAP10k_contigs.sam\t (format: SAM)\n",
      "Thread #0: \t50000 reads finished. 2 secs passed\n",
      "Thread #1: \t100000 reads finished. 2 secs passed\n",
      "Thread #2: \t150000 reads finished. 3 secs passed\n",
      "Thread #6: \t200000 reads finished. 3 secs passed\n",
      "Thread #7: \t250000 reads finished. 3 secs passed\n",
      "Thread #3: \t300000 reads finished. 3 secs passed\n",
      "Thread #4: \t350000 reads finished. 3 secs passed\n",
      "Thread #5: \t400000 reads finished. 3 secs passed\n",
      "Thread #0: \t450000 reads finished. 3 secs passed\n",
      "Thread #1: \t500000 reads finished. 3 secs passed\n",
      "Thread #2: \t550000 reads finished. 4 secs passed\n",
      "Thread #6: \t600000 reads finished. 4 secs passed\n",
      "Thread #7: \t650000 reads finished. 4 secs passed\n",
      "Thread #3: \t700000 reads finished. 4 secs passed\n",
      "Thread #4: \t750000 reads finished. 4 secs passed\n",
      "Thread #5: \t800000 reads finished. 4 secs passed\n",
      "Thread #0: \t850000 reads finished. 4 secs passed\n",
      "Thread #1: \t900000 reads finished. 5 secs passed\n",
      "Thread #2: \t950000 reads finished. 5 secs passed\n",
      "Thread #6: \t1000000 reads finished. 5 secs passed\n",
      "Thread #7: \t1050000 reads finished. 5 secs passed\n",
      "Thread #3: \t1100000 reads finished. 5 secs passed\n",
      "Thread #4: \t1150000 reads finished. 5 secs passed\n",
      "Thread #5: \t1200000 reads finished. 5 secs passed\n",
      "Thread #0: \t1250000 reads finished. 5 secs passed\n",
      "Thread #1: \t1300000 reads finished. 6 secs passed\n",
      "Thread #2: \t1350000 reads finished. 6 secs passed\n",
      "Thread #6: \t1400000 reads finished. 6 secs passed\n",
      "Thread #7: \t1450000 reads finished. 6 secs passed\n",
      "Thread #3: \t1500000 reads finished. 6 secs passed\n",
      "Thread #4: \t1550000 reads finished. 6 secs passed\n",
      "Thread #5: \t1600000 reads finished. 6 secs passed\n",
      "Thread #0: \t1650000 reads finished. 6 secs passed\n",
      "Thread #1: \t1700000 reads finished. 7 secs passed\n",
      "Thread #2: \t1750000 reads finished. 7 secs passed\n",
      "Thread #6: \t1800000 reads finished. 8 secs passed\n",
      "Thread #7: \t1850000 reads finished. 8 secs passed\n",
      "Thread #3: \t1900000 reads finished. 8 secs passed\n",
      "Thread #4: \t1950000 reads finished. 9 secs passed\n",
      "Thread #5: \t2000000 reads finished. 9 secs passed\n",
      "Thread #0: \t2050000 reads finished. 9 secs passed\n",
      "Thread #1: \t2100000 reads finished. 9 secs passed\n",
      "Thread #2: \t2150000 reads finished. 9 secs passed\n",
      "Thread #6: \t2200000 reads finished. 9 secs passed\n",
      "Thread #7: \t2250000 reads finished. 10 secs passed\n",
      "Thread #3: \t2300000 reads finished. 10 secs passed\n",
      "Thread #4: \t2350000 reads finished. 10 secs passed\n",
      "Thread #5: \t2400000 reads finished. 10 secs passed\n",
      "Thread #0: \t2450000 reads finished. 10 secs passed\n",
      "Thread #1: \t2500000 reads finished. 10 secs passed\n",
      "Thread #2: \t2550000 reads finished. 10 secs passed\n",
      "Thread #6: \t2600000 reads finished. 10 secs passed\n",
      "Thread #7: \t2650000 reads finished. 11 secs passed\n",
      "Thread #3: \t2700000 reads finished. 11 secs passed\n",
      "Thread #4: \t2750000 reads finished. 11 secs passed\n",
      "Thread #5: \t2800000 reads finished. 11 secs passed\n",
      "Thread #0: \t2850000 reads finished. 11 secs passed\n",
      "Thread #1: \t2900000 reads finished. 11 secs passed\n",
      "Thread #2: \t2950000 reads finished. 11 secs passed\n",
      "Thread #6: \t3000000 reads finished. 12 secs passed\n",
      "Thread #7: \t3050000 reads finished. 12 secs passed\n",
      "Thread #3: \t3100000 reads finished. 12 secs passed\n",
      "Thread #4: \t3150000 reads finished. 12 secs passed\n",
      "Thread #5: \t3200000 reads finished. 12 secs passed\n",
      "Thread #0: \t3250000 reads finished. 12 secs passed\n",
      "Thread #1: \t3300000 reads finished. 12 secs passed\n",
      "Thread #2: \t3350000 reads finished. 13 secs passed\n",
      "Thread #6: \t3400000 reads finished. 13 secs passed\n",
      "Thread #7: \t3450000 reads finished. 13 secs passed\n",
      "Thread #3: \t3500000 reads finished. 13 secs passed\n",
      "Thread #4: \t3550000 reads finished. 13 secs passed\n",
      "Thread #5: \t3600000 reads finished. 13 secs passed\n",
      "Thread #0: \t3650000 reads finished. 13 secs passed\n",
      "Thread #1: \t3700000 reads finished. 13 secs passed\n",
      "Thread #2: \t3750000 reads finished. 14 secs passed\n",
      "Thread #6: \t3800000 reads finished. 14 secs passed\n",
      "Thread #7: \t3850000 reads finished. 14 secs passed\n",
      "Thread #3: \t3900000 reads finished. 14 secs passed\n",
      "Thread #4: \t3950000 reads finished. 14 secs passed\n",
      "Thread #5: \t4000000 reads finished. 14 secs passed\n",
      "Thread #0: \t4050000 reads finished. 14 secs passed\n",
      "Thread #1: \t4100000 reads finished. 14 secs passed\n",
      "Thread #2: \t4150000 reads finished. 15 secs passed\n",
      "Thread #6: \t4200000 reads finished. 15 secs passed\n",
      "Thread #7: \t4250000 reads finished. 15 secs passed\n",
      "Thread #3: \t4300000 reads finished. 15 secs passed\n",
      "Thread #4: \t4350000 reads finished. 15 secs passed\n",
      "Thread #5: \t4400000 reads finished. 15 secs passed\n",
      "Thread #0: \t4450000 reads finished. 15 secs passed\n",
      "Thread #1: \t4500000 reads finished. 15 secs passed\n",
      "Thread #2: \t4550000 reads finished. 16 secs passed\n",
      "Thread #6: \t4600000 reads finished. 16 secs passed\n",
      "Thread #7: \t4650000 reads finished. 16 secs passed\n",
      "Thread #3: \t4700000 reads finished. 16 secs passed\n",
      "Thread #4: \t4750000 reads finished. 16 secs passed\n",
      "Thread #5: \t4800000 reads finished. 16 secs passed\n",
      "Thread #0: \t4850000 reads finished. 16 secs passed\n",
      "Thread #1: \t4900000 reads finished. 17 secs passed\n",
      "Thread #2: \t4950000 reads finished. 17 secs passed\n",
      "Thread #6: \t5000000 reads finished. 17 secs passed\n",
      "Thread #7: \t5050000 reads finished. 17 secs passed\n",
      "Thread #3: \t5100000 reads finished. 17 secs passed\n",
      "Thread #4: \t5150000 reads finished. 17 secs passed\n",
      "Thread #5: \t5200000 reads finished. 17 secs passed\n",
      "Thread #0: \t5250000 reads finished. 18 secs passed\n",
      "Thread #1: \t5300000 reads finished. 18 secs passed\n",
      "Thread #2: \t5350000 reads finished. 18 secs passed\n",
      "Thread #6: \t5400000 reads finished. 18 secs passed\n",
      "Thread #7: \t5450000 reads finished. 18 secs passed\n",
      "Thread #3: \t5500000 reads finished. 18 secs passed\n",
      "Thread #4: \t5550000 reads finished. 18 secs passed\n",
      "Thread #5: \t5600000 reads finished. 18 secs passed\n",
      "Thread #0: \t5650000 reads finished. 19 secs passed\n",
      "Thread #1: \t5700000 reads finished. 19 secs passed\n",
      "Thread #2: \t5750000 reads finished. 19 secs passed\n",
      "Thread #6: \t5800000 reads finished. 19 secs passed\n",
      "Thread #7: \t5850000 reads finished. 19 secs passed\n",
      "Thread #3: \t5900000 reads finished. 19 secs passed\n",
      "Thread #4: \t5950000 reads finished. 19 secs passed\n",
      "Thread #5: \t6000000 reads finished. 19 secs passed\n",
      "Thread #0: \t6050000 reads finished. 20 secs passed\n",
      "Thread #1: \t6100000 reads finished. 20 secs passed\n",
      "Thread #2: \t6150000 reads finished. 20 secs passed\n",
      "Thread #6: \t6200000 reads finished. 20 secs passed\n",
      "Thread #7: \t6250000 reads finished. 20 secs passed\n",
      "Thread #3: \t6300000 reads finished. 20 secs passed\n",
      "Thread #4: \t6350000 reads finished. 20 secs passed\n",
      "Thread #5: \t6400000 reads finished. 21 secs passed\n",
      "Thread #0: \t6450000 reads finished. 21 secs passed\n",
      "Thread #1: \t6500000 reads finished. 21 secs passed\n",
      "Thread #2: \t6550000 reads finished. 21 secs passed\n",
      "Thread #6: \t6600000 reads finished. 21 secs passed\n",
      "Thread #7: \t6650000 reads finished. 21 secs passed\n",
      "Thread #3: \t6700000 reads finished. 21 secs passed\n",
      "Thread #4: \t6750000 reads finished. 21 secs passed\n",
      "Thread #5: \t6800000 reads finished. 22 secs passed\n",
      "Thread #0: \t6850000 reads finished. 22 secs passed\n",
      "Thread #1: \t6900000 reads finished. 22 secs passed\n",
      "Thread #2: \t6950000 reads finished. 22 secs passed\n",
      "Thread #6: \t7000000 reads finished. 23 secs passed\n",
      "Thread #7: \t7050000 reads finished. 23 secs passed\n",
      "Thread #3: \t7100000 reads finished. 23 secs passed\n",
      "Thread #4: \t7150000 reads finished. 23 secs passed\n",
      "Thread #5: \t7200000 reads finished. 23 secs passed\n",
      "Thread #0: \t7250000 reads finished. 23 secs passed\n",
      "Thread #1: \t7300000 reads finished. 23 secs passed\n",
      "Thread #2: \t7350000 reads finished. 23 secs passed\n",
      "Thread #6: \t7400000 reads finished. 24 secs passed\n",
      "Thread #7: \t7450000 reads finished. 24 secs passed\n",
      "Thread #3: \t7500000 reads finished. 24 secs passed\n",
      "Thread #4: \t7550000 reads finished. 24 secs passed\n",
      "Thread #5: \t7600000 reads finished. 24 secs passed\n",
      "Thread #0: \t7650000 reads finished. 24 secs passed\n",
      "Thread #1: \t7700000 reads finished. 24 secs passed\n",
      "Thread #2: \t7750000 reads finished. 24 secs passed\n",
      "Thread #6: \t7800000 reads finished. 25 secs passed\n",
      "Thread #7: \t7850000 reads finished. 25 secs passed\n",
      "Thread #3: \t7900000 reads finished. 25 secs passed\n",
      "Thread #4: \t7950000 reads finished. 25 secs passed\n",
      "Thread #5: \t8000000 reads finished. 25 secs passed\n",
      "Thread #0: \t8050000 reads finished. 25 secs passed\n",
      "Thread #1: \t8100000 reads finished. 25 secs passed\n",
      "Thread #2: \t8150000 reads finished. 26 secs passed\n",
      "Thread #6: \t8200000 reads finished. 26 secs passed\n",
      "Thread #7: \t8250000 reads finished. 26 secs passed\n",
      "Thread #3: \t8300000 reads finished. 26 secs passed\n",
      "Thread #4: \t8350000 reads finished. 26 secs passed\n",
      "Thread #5: \t8400000 reads finished. 26 secs passed\n",
      "Thread #0: \t8450000 reads finished. 26 secs passed\n",
      "Thread #1: \t8500000 reads finished. 26 secs passed\n",
      "Thread #2: \t8550000 reads finished. 27 secs passed\n",
      "Thread #6: \t8600000 reads finished. 27 secs passed\n",
      "Thread #7: \t8650000 reads finished. 27 secs passed\n",
      "Thread #3: \t8700000 reads finished. 27 secs passed\n",
      "Thread #4: \t8750000 reads finished. 27 secs passed\n",
      "Thread #5: \t8800000 reads finished. 27 secs passed\n",
      "Thread #0: \t8850000 reads finished. 27 secs passed\n",
      "Thread #1: \t8900000 reads finished. 27 secs passed\n",
      "Thread #2: \t8950000 reads finished. 28 secs passed\n",
      "Thread #6: \t9000000 reads finished. 28 secs passed\n",
      "Thread #7: \t9050000 reads finished. 28 secs passed\n",
      "Thread #3: \t9100000 reads finished. 28 secs passed\n",
      "Thread #4: \t9150000 reads finished. 29 secs passed\n",
      "Thread #5: \t9200000 reads finished. 29 secs passed\n",
      "Thread #0: \t9250000 reads finished. 30 secs passed\n",
      "Thread #7: \t9402890 reads finished. 30 secs passed\n",
      "Thread #1: \t9300000 reads finished. 30 secs passed\n",
      "Thread #2: \t9350000 reads finished. 30 secs passed\n",
      "Thread #6: \t9400000 reads finished. 30 secs passed\n",
      "Total number of aligned reads: 348658 (3.7%)\n",
      "Done.\n",
      "Finished at Fri Jan 22 12:24:15 2016\n",
      "Total time consumed:  30 secs\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/BSMAP/bsmap-2.74/bsmap \\\n",
    "-a /Volumes/web-1/nightingales/O_lurida/1_ATCACG_L001_R1_001.fastq.gz \\\n",
    "-d /Volumes/Histidine/hectocotylus/OlyO_Pat_PacBio_10k_contigs.fa \\\n",
    "-o /Volumes/Histidine/hectocotylus/1_ATCACG_BSMAP10k_contigs.sam \\\n",
    "-p 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  File \"/Applications/bioinfo/BSMAP/bsmap-2.74/methratio.py\", line 227\r\n",
      "    if nc > 0: print 'total %d valid mappings, %d covered cytosines, average coverage: %.2f fold.' % (nmap, nc, float(nd)/nc)\r\n",
      "                                                                                                 ^\r\n",
      "SyntaxError: invalid syntax\r\n"
     ]
    }
   ],
   "source": [
    "!python /Applications/bioinfo/BSMAP/bsmap-2.74/methratio.py \\\n",
    "-d /Volumes/Histidine/hectocotylus/OlyO_Pat_PacBio_10k_contigs.fa \\\n",
    "-o /Volumes/Histidine/hectocotylus/1_ATCACG_BSMAP10k_contigs_methratio.txt \\\n",
    "-s /Applications/bioinfo/BSMAP/bsmap-2.74/samtools \\\n",
    "/Volumes/Histidine/hectocotylus/1_ATCACG_BSMAP10k_contigs.sam "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  File \"/Applications/bioinfo/BSMAP/bsmap-2.74/methratio.py\", line 227\r\n",
      "    if nc > 0: print 'total %d valid mappings, %d covered cytosines, average coverage: %.2f fold.' % (nmap, nc, float(nd)/nc)\r\n",
      "                                                                                                 ^\r\n",
      "SyntaxError: invalid syntax\r\n"
     ]
    }
   ],
   "source": [
    "!python /Applications/bioinfo/BSMAP/bsmap-2.74/methratio.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!/Applications/bioinfo/BSMAP/bsmap-2.74/bsmap \\\n",
    "-a /Volumes/web-1/nightingales/O_lurida/1_ATCACG_L001_R1_001.fastq.gz \\\n",
    "-d /Volumes/Histidine/hectocotylus/OlyO_Pat_PacBio_10k_contigs.fa \\\n",
    "-o /Volumes/Histidine/hectocotylus/1_ATCACG_BSMAP10k_contigs.sam \\\n",
    "-p 8"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
