{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/ICSASG_v2_-_Assembly_-_NCBI_1CDBE816.png\" alt=\"ICSASG_v2_-_Assembly_-_NCBI_1CDBE816.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Index_of__genomes_all_GCA_000233375_4_ICSASG_v2_1CDBE833.png\" alt=\"Index_of__genomes_all_GCA_000233375_4_ICSASG_v2_1CDBE833.png\"/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/git-repos/nb-2016/S_salar\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web-1/halfshell/working-directory\n",
      "/Volumes/web-1/halfshell/working-directory\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/web-1/halfshell/working-directory\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir 16-05-05\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web-1/halfshell/working-directory/16-05-05\n",
      "/Volumes/web-1/halfshell/working-directory/16-05-05\n"
     ]
    }
   ],
   "source": [
    "cd 16-05-05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000233375.4_ICSASG_v2_assembly_report.txt GCA_000233375.4_ICSASG_v2_genomic.gff.gz      README.txt\n",
      "GCA_000233375.4_ICSASG_v2_assembly_stats.txt  GCA_000233375.4_ICSASG_v2_rm.out.gz           download\n",
      "GCA_000233375.4_ICSASG_v2_genomic.fna.gz      GCA_000233375.4_ICSASG_v2_rm.run\n",
      "GCA_000233375.4_ICSASG_v2_genomic.gbff.gz     GCA_000233375.4_ICSASG_v2_wgsmaster.gbff.gz\n",
      "GCA_000233375.4_ICSASG_v2_assembly_report.txt GCA_000233375.4_ICSASG_v2_genomic.gff.gz      README.txt\n",
      "GCA_000233375.4_ICSASG_v2_assembly_stats.txt  GCA_000233375.4_ICSASG_v2_rm.out.gz           download\n",
      "GCA_000233375.4_ICSASG_v2_genomic.fna.gz      GCA_000233375.4_ICSASG_v2_rm.run\n",
      "GCA_000233375.4_ICSASG_v2_genomic.gbff.gz     GCA_000233375.4_ICSASG_v2_wgsmaster.gbff.gz\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "################################################################################\n",
      "README for ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/\n",
      "           ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/\n",
      "           ftp://ftp.ncbi.nlm.nih.gov/genomes/all/\n",
      "\n",
      "Last updated: March 14, 2016\n",
      "################################################################################\n",
      "\n",
      "==========\n",
      "Background\n",
      "==========\n",
      "Sequence data is provided for all single organism genome assemblies that are \n",
      "included in NCBI's Assembly resource (www.ncbi.nlm.nih.gov/assembly/).  This \n",
      "includes submissions to databases of the International Nucleotide Sequence \n",
      "Database Collaboration, which are available in NCBI's GenBank database, as well \n",
      "as the subset of those submissions that are included in NCBI's RefSeq Genomes \n",
      "project. \n",
      "\n",
      "Available by anonymous FTP at:\n",
      "\n",
      "        ftp://ftp.ncbi.nlm.nih.gov/genomes/\n",
      "\n",
      "Please refer to README files and the FTP FAQ for additional information:\n",
      "       http://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/\n",
      "\n",
      "\n",
      "=====================================================================\n",
      "Genome sequence and annotation data is provided in three directories:\n",
      "=====================================================================\n",
      "1) all:     content is the union of GenBank and RefSeq assemblies. \n",
      "            Subdirectories are provided per assembly accession and version.\n",
      "            [This directory is not suitable for browsing because it holds many \n",
      "            thousands of entries.] Only subdirectories for \"latest\" assemblies\n",
      "            are refreshed when annotation is updated or when software updates\n",
      "            are released, so new file formats or improvements to existing \n",
      "            formats are not available for non-latest assemblies.\n",
      "2) genbank: content includes primary submissions of assembled genome sequence \n",
      "            and associated annotation data, if any, as exchanged among members \n",
      "            of the International Nucleotide Sequence Database Collaboration, \n",
      "            of which NCBI's GenBank database is a member. The GenBank directory \n",
      "            area includes genome sequence data for a larger number of organisms \n",
      "            than the RefSeq directory area; however, some assemblies are \n",
      "            unannotated. The sub-directory structure includes:\n",
      "            a. archaea\n",
      "            b. bacteria\n",
      "            c. fungi\n",
      "            d. invertebrate\n",
      "            e. other -  this directory includes synthetic genomes\n",
      "            f. plant\n",
      "            g. protozoa\n",
      "            h. vertebrate_mammalian\n",
      "            i. vertebrate_other\n",
      "3) refseq:  content includes assembled genome sequence and RefSeq annotation \n",
      "            data. All prokaryotic and eukaryotic RefSeq genomes have annotation. \n",
      "            RefSeq annotation data may be calculated by NCBI annotation  \n",
      "            pipelines or propagated from the GenBank submission. The RefSeq \n",
      "            directory area includes fewer organisms than the GenBank directory\n",
      "            area because not all genome assemblies are selected for the RefSeq\n",
      "            project. \n",
      "            Sub-directories include:\n",
      "            a. archaea\n",
      "            b. bacteria\n",
      "            c. fungi\n",
      "            d. invertebrate\n",
      "            e. plant\n",
      "            f. protozoa\n",
      "            g. vertebrate_mammalian\n",
      "            h. vertebrate_other \n",
      "            i. viral\n",
      "            j. mitochondrion [Content of the mitochondrion, plasmid and plastid\n",
      "            k. plasmid     directories is from the RefSeq release FTP site. See \n",
      "            l. plastid     ftp://ftp.ncbi.nlm.nih.gov/refseq/release/README]\n",
      "\n",
      "Data are further organized within each of the above directories as a series of \n",
      "directories named as the species binomial. For example:\n",
      "   ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/Escherichia_coli/\n",
      "           - or - \n",
      "   ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/\n",
      "\n",
      "The next hierarchy provides access to all assemblies for the species, latest \n",
      "assemblies, and selected reference or representative assemblies for the species \n",
      "(if any). Within these groupings, sequence and annotation (and other) data is \n",
      "provided per assembly in a series of directories that are named using the rule:\n",
      "\n",
      "   [Assembly accession.version]_[assembly name]\n",
      "\n",
      "For example, the directory hierarchy for the GenBank Escherichia coli K-12 \n",
      "subst. MG1655 genome, which has the assembly accession GCA_000005845.2 and \n",
      "default assembly name ASM584v2 looks like this:  \n",
      "   /genomes/genbank/bacteria/Escherichia_coli/all_assembly_versions/GCA_000005845.2_ASM584v2  \n",
      "\n",
      "The directory hierarchy for the RefSeq annotated human reference genome which \n",
      "has the assembly accession GCF_000001405.30 and assembly name GRCh38.p4 looks \n",
      "like this:\n",
      "   /genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.30_GRCh38.p4\n",
      "\n",
      "Genome assemblies of interest can be identified using the NCBI Assembly resource\n",
      "(www.ncbi.nlm.nih.gov/assembly), or by using the assembly summary report files \n",
      "that are provided for both all genbank and all refseq assemblies:\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/ASSEMBLY_REPORTS/assembly_summary_genbank.txt\n",
      "or ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly_summary_genbank.txt\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/ASSEMBLY_REPORTS/assembly_summary_refseq.txt\n",
      "or ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/assembly_summary_refseq.txt\n",
      "\n",
      "Assembly summary report files containing information on assemblies for a \n",
      "particular taxonomic group or species are provided in the group and \n",
      "Genus_species directories under the \"genbank\" and \"refseq\" directory trees. e.g.\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/Sulfolobus_islandicus/assembly_summary.txt\n",
      "\n",
      "Search the meta-data fields, or filter the files, to find assemblies of \n",
      "interest.\n",
      "\n",
      "\n",
      "===========================\n",
      "Data provided per assembly:\n",
      "===========================\n",
      "Sequence and other data files provided per assembly are named according to the \n",
      "rule:\n",
      "[assembly accession.version]_[assembly name]_[content type].[optional format]\n",
      "\n",
      "File formats and content:\n",
      "\n",
      "   *_assembly_report.txt file\n",
      "       Tab-delimited text file reporting the name, role and sequence \n",
      "       accession.version for objects in the assembly. The file header contains \n",
      "       meta-data for the assembly including: assembly name, assembly \n",
      "       accession.version, scientific name of the organism and its taxonomy ID, \n",
      "       assembly submitter, and sequence release date. The file header also \n",
      "       indicates whether the assembly version is \"latest\", \"replaced\", or \n",
      "       \"suppressed\".\n",
      "   *_assembly_stats.txt file\n",
      "       Tab-delimited text file reporting statistics for the assembly including: \n",
      "       total length, ungapped length, contig & scaffold counts, contig-N50, \n",
      "       scaffold-L50, scaffold-N50, scaffold-N75, and scaffold-N90\n",
      "   *_assembly_regions.txt\n",
      "       Provided for assemblies that include alternate or patch assembly units. \n",
      "       Tab-delimited text file reporting the location of genomic regions and \n",
      "       the alt/patch scaffolds placed within those regions.\n",
      "   *_assembly_structure directory\n",
      "       This directory will only be present if the assembly has internal \n",
      "       structure. When present, it will contain AGP files that define how \n",
      "       component sequences are organized into scaffolds and/or chromosomes. \n",
      "       Other files define how scaffolds and chromosomes are organized into \n",
      "       non-nuclear and other assembly-units, and how any alternate or patch \n",
      "       scaffolds are placed relative to the chromosomes. Refer to the README.txt\n",
      "       file in the assembly_structure directory for additional information.\n",
      "   *_feature_table.txt.gz\n",
      "       Tab-delimited text file reporting locations and attributes for a subset \n",
      "       of annotated features. Included feature types are: gene, CDS, RNA (all \n",
      "       types), operon, C/V/N/S_region, and V/D/J_segment. Replaces the .ptt & \n",
      "       .rnt format files that were provided in the old genomes FTP directories.\n",
      "       See the \"Description of files\" section below for details of the file \n",
      "       format.\n",
      "   *_genomic.fna.gz file\n",
      "       FASTA format of the genomic sequence(s) in the assembly. Repetitive \n",
      "       sequences in eukaryotes are masked to lower-case (see below).\n",
      "       The FASTA title is formatted as sequence accession.version plus \n",
      "       description. The genomic.fna.gz file includes all top-level sequences in\n",
      "       the assembly (chromosomes, plasmids, organelles, unlocalized scaffolds,\n",
      "       unplaced scaffolds, and any alternate loci or patch scaffolds). Scaffolds\n",
      "       that are part of the chromosomes are not included because they are\n",
      "       redundant with the chromosome sequences; sequences for these placed \n",
      "       scaffolds are provided under the assembly_structure directory.\n",
      "   *_genomic.gbff.gz file\n",
      "       GenBank flat file format of the genomic sequence(s) in the assembly. This\n",
      "       file includes both the genomic sequence and the CONTIG description (for \n",
      "       CON records), hence, it replaces both the .gbk & .gbs format files that \n",
      "       were provided in the old genomes FTP directories.\n",
      "   *_genomic.gff.gz file\n",
      "       Annotation of the genomic sequence(s) in Generic Feature Format Version 3\n",
      "       (GFF3). Sequence identifiers are provided as accession.version.\n",
      "       Additional information about NCBI's GFF files is available at \n",
      "       ftp://ftp.ncbi.nlm.nih.gov/genomes/README_GFF3.txt.\n",
      "   *_protein.faa.gz file\n",
      "       FASTA format of the accessioned protein products annotated on the genome \n",
      "       assembly\n",
      "       The FASTA title is formatted as sequence accession.version plus \n",
      "       description.\n",
      "   *_protein.gpff.gz file\n",
      "       GenPept format of the accessioned protein products annotated on the \n",
      "       genome assembly\n",
      "   *_rm.out.gz file\n",
      "       RepeatMasker output; \n",
      "       Provided for Eukaryotes \n",
      "   *_rm.run file\n",
      "       Documentation of the RepeatMasker version, parameters, and library; \n",
      "       Provided for Eukaryotes \n",
      "   *_rna.fna.gz file\n",
      "       FASTA format of accessioned RNA products annotated on the genome \n",
      "       assembly; Provided for RefSeq assemblies as relevant (Note, RNA and mRNA \n",
      "       products are not instantiated as a separate accessioned record in GenBank\n",
      "       but are provided for some RefSeq genomes, most notably the eukaryotes.)\n",
      "       The FASTA title is provided as sequence accession.version plus \n",
      "       description.\n",
      "   *_rna.gbff.gz file\n",
      "       GenBank flat file format of RNA products annotated on the genome \n",
      "       assembly; Provided for RefSeq assemblies as relevant\n",
      "   *_wgsmaster.gbff.gz\n",
      "       GenBank flat file format of the WGS master for the assembly (present only\n",
      "       if a WGS master record exists for the sequences in the assembly).\n",
      "   md5checksums.txt file\n",
      "       file checksums are provided for all data files in the directory\n",
      "\n",
      "\n",
      "=====================\n",
      "Description of files:\n",
      "=====================\n",
      "\n",
      "Masking of fasta sequences in genomic.fna.gz files\n",
      "--------------------------------------------------\n",
      "Repetitive sequences in eukaryotic genome assembly sequence files, as \n",
      "identified by WindowMasker (Morgulis A, Gertz EM, Schaffer AA, Agarwala R. \n",
      "2006. Bioinformatics 22:134-41), have been masked to lower-case.\n",
      "\n",
      "Alignment programs typically have parameters that control whether the program \n",
      "will ignore lower-case masking, treat it as soft-masking (i.e. only for finding \n",
      "initial matches) or treat it as hard-masking. By default NCBI BLAST will ignore \n",
      "lower-case masking but this can be changed by adding options to the blastn \n",
      "command-line.\n",
      "To have blastn treat lower-case masking in the query sequence as soft-masking \n",
      "add:\n",
      "     -lcase_masking\n",
      "To have blastn treat lower-case masking in the query sequence as hard-masking \n",
      "add:\n",
      "     -lcase_masking -soft_masking false\n",
      "\n",
      "Alternatively, commands such as the following can be used to generate either \n",
      "unmasked sequence or sequence masked with Ns.\n",
      "\n",
      "Example commands to remove lower-case masking:\n",
      "perl -pe '/^[^>]/ and $_=uc' genomic.fna > genomic.unmasked.fna\n",
      "  -or-\n",
      "awk '{if(/^[^>]/)$0=toupper($0);print $0}' genomic.fna > genomic.unmasked.fna\n",
      "\n",
      "Example commands to convert lower-case masking to masking with Ns (hard-masked):\n",
      "perl -pe '/^[^>]/ and $_=~ s/[a-z]/N/g' genomic.fna > genomic.N-masked.fna\n",
      "  -or-\n",
      "awk '{if(/^[^>]/)gsub(/[a-z]/,\"N\");print $0}' genomic.fna > genomic.N-masked.fna\n",
      "\n",
      "\n",
      "*_feature_table.txt.gz\n",
      "----------------------\n",
      "Tab-delimited text file reporting locations and attributes for a subset of \n",
      "annotated features. Included feature types are: gene, CDS, RNA (all types), \n",
      "operon, C/V/N/S_region, and V/D/J_segment. \n",
      "\n",
      "The file is tab delimited (including a #header) with the following columns:\n",
      "col 1: feature: INSDC feature type\n",
      "col 2: class: For ncRNA features, this is the ncRNA_class for the feature. For \n",
      "       gene features, this is the gene biotype computed based on the set of \n",
      "       child features for that gene. See the description of the gene_biotype \n",
      "       attribute in the GFF3 documentation for more details:\n",
      "       ftp://ftp.ncbi.nlm.nih.gov/genomes/README_GFF3.txt\n",
      "col 3: assembly: assembly accession.version\n",
      "col 4: assembly_unit: name of the assembly unit, such as \"Primary Assembly\", \n",
      "       \"ALT_REF_LOCI_1\", or \"non-nuclear\"\n",
      "col 5: seq_type: sequence type, computed from the \"Sequence-Role\" and \n",
      "       \"Assigned-Molecule-Location/Type\" in the *_assembly_report.txt file. The\n",
      "       value is computed as:\n",
      "       if an assembled-molecule, then reports the location/type value. e.g. \n",
      "       chromosome, mitochondrion, or plasmid\n",
      "       if an unlocalized-scaffold, then report \"unlocalized scaffold on <type>\".\n",
      "       e.g. unlocalized scaffold on chromosome\n",
      "       else the role, e.g. alternate scaffold, fix patch, or novel patch\n",
      "col 6: chromosome\n",
      "col 7: genomic_accession\n",
      "col 8: start: feature start coordinate (base-1). start is always less than end\n",
      "col 9: end: feature end coordinate (base-1)\n",
      "col10: strand\n",
      "col11: product_accession: accession.version of the product referenced by this \n",
      "       feature, if exists\n",
      "col12: non-redundant_refseq: for bacteria and archaea assemblies, the \n",
      "       non-redundant WP_ protein accession corresponding to the CDS feature. May\n",
      "       be the same as column 11, for RefSeq genomes annotated directly with WP_\n",
      "       RefSeq proteins, or may be different, for genomes annotated with \n",
      "       genome-specific protein accessions (e.g. NP_ or YP_ RefSeq proteins) that\n",
      "       reference a WP_ RefSeq accession.\n",
      "col13: related_accession: for eukaryotic RefSeq annotations, the RefSeq protein\n",
      "       accession corresponding to the transcript feature, or the RefSeq \n",
      "       transcript accession corresponding to the protein feature.\n",
      "col14: name: For genes, this is the gene description or full name. For RNA, CDS,\n",
      "       and some other features, this is the product name.\n",
      "col15: symbol: gene symbol\n",
      "col16: GeneID: NCBI GeneID, for those RefSeq genomes included in NCBI's Gene \n",
      "       resource\n",
      "col17: locus_tag\n",
      "col18: feature_interval_length: sum of the lengths of all intervals for the \n",
      "       feature (i.e. the length without introns for a joined feature)\n",
      "col19: product_length: length of the product corresponding to the \n",
      "       accession.version in column 11. Protein product lengths are in amino acid\n",
      "       units, and do not include the stop codon which is included in column 18.\n",
      "       Additionally, product_length may differ from feature_interval_length if \n",
      "       the product contains sequence differences vs. the genome, as found for \n",
      "       some RefSeq transcript and protein products based on mRNA sequences and \n",
      "       also for INSDC proteins that are submitted to correct genome \n",
      "       discrepancies.\n",
      "col20: attributes: semi-colon delimited list of a controlled set of qualifiers.\n",
      "       The list currently includes:\n",
      "       partial, pseudo, pseudogene, ribosomal_slippage, trans_splicing, \n",
      "       anticodon=NNN (for tRNAs), old_locus_tag=XXX \n",
      "\n",
      "________________________________________________________________________________\n",
      "National Center for Biotechnology Information (NCBI)\n",
      "National Library of Medicine\n",
      "National Institutes of Health\n",
      "8600 Rockville Pike\n",
      "Bethesda, MD 20894, USA\n",
      "tel: (301) 496-2475\n",
      "fax: (301) 480-9241\n",
      "e-mail: info@ncbi.nlm.nih.gov\n",
      "________________________________________________________________________________\n",
      "################################################################################\n",
      "README for ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/\n",
      "           ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/\n",
      "           ftp://ftp.ncbi.nlm.nih.gov/genomes/all/\n",
      "\n",
      "Last updated: March 14, 2016\n",
      "################################################################################\n",
      "\n",
      "==========\n",
      "Background\n",
      "==========\n",
      "Sequence data is provided for all single organism genome assemblies that are \n",
      "included in NCBI's Assembly resource (www.ncbi.nlm.nih.gov/assembly/).  This \n",
      "includes submissions to databases of the International Nucleotide Sequence \n",
      "Database Collaboration, which are available in NCBI's GenBank database, as well \n",
      "as the subset of those submissions that are included in NCBI's RefSeq Genomes \n",
      "project. \n",
      "\n",
      "Available by anonymous FTP at:\n",
      "\n",
      "        ftp://ftp.ncbi.nlm.nih.gov/genomes/\n",
      "\n",
      "Please refer to README files and the FTP FAQ for additional information:\n",
      "       http://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/\n",
      "\n",
      "\n",
      "=====================================================================\n",
      "Genome sequence and annotation data is provided in three directories:\n",
      "=====================================================================\n",
      "1) all:     content is the union of GenBank and RefSeq assemblies. \n",
      "            Subdirectories are provided per assembly accession and version.\n",
      "            [This directory is not suitable for browsing because it holds many \n",
      "            thousands of entries.] Only subdirectories for \"latest\" assemblies\n",
      "            are refreshed when annotation is updated or when software updates\n",
      "            are released, so new file formats or improvements to existing \n",
      "            formats are not available for non-latest assemblies.\n",
      "2) genbank: content includes primary submissions of assembled genome sequence \n",
      "            and associated annotation data, if any, as exchanged among members \n",
      "            of the International Nucleotide Sequence Database Collaboration, \n",
      "            of which NCBI's GenBank database is a member. The GenBank directory \n",
      "            area includes genome sequence data for a larger number of organisms \n",
      "            than the RefSeq directory area; however, some assemblies are \n",
      "            unannotated. The sub-directory structure includes:\n",
      "            a. archaea\n",
      "            b. bacteria\n",
      "            c. fungi\n",
      "            d. invertebrate\n",
      "            e. other -  this directory includes synthetic genomes\n",
      "            f. plant\n",
      "            g. protozoa\n",
      "            h. vertebrate_mammalian\n",
      "            i. vertebrate_other\n",
      "3) refseq:  content includes assembled genome sequence and RefSeq annotation \n",
      "            data. All prokaryotic and eukaryotic RefSeq genomes have annotation. \n",
      "            RefSeq annotation data may be calculated by NCBI annotation  \n",
      "            pipelines or propagated from the GenBank submission. The RefSeq \n",
      "            directory area includes fewer organisms than the GenBank directory\n",
      "            area because not all genome assemblies are selected for the RefSeq\n",
      "            project. \n",
      "            Sub-directories include:\n",
      "            a. archaea\n",
      "            b. bacteria\n",
      "            c. fungi\n",
      "            d. invertebrate\n",
      "            e. plant\n",
      "            f. protozoa\n",
      "            g. vertebrate_mammalian\n",
      "            h. vertebrate_other \n",
      "            i. viral\n",
      "            j. mitochondrion [Content of the mitochondrion, plasmid and plastid\n",
      "            k. plasmid     directories is from the RefSeq release FTP site. See \n",
      "            l. plastid     ftp://ftp.ncbi.nlm.nih.gov/refseq/release/README]\n",
      "\n",
      "Data are further organized within each of the above directories as a series of \n",
      "directories named as the species binomial. For example:\n",
      "   ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/Escherichia_coli/\n",
      "           - or - \n",
      "   ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/\n",
      "\n",
      "The next hierarchy provides access to all assemblies for the species, latest \n",
      "assemblies, and selected reference or representative assemblies for the species \n",
      "(if any). Within these groupings, sequence and annotation (and other) data is \n",
      "provided per assembly in a series of directories that are named using the rule:\n",
      "\n",
      "   [Assembly accession.version]_[assembly name]\n",
      "\n",
      "For example, the directory hierarchy for the GenBank Escherichia coli K-12 \n",
      "subst. MG1655 genome, which has the assembly accession GCA_000005845.2 and \n",
      "default assembly name ASM584v2 looks like this:  \n",
      "   /genomes/genbank/bacteria/Escherichia_coli/all_assembly_versions/GCA_000005845.2_ASM584v2  \n",
      "\n",
      "The directory hierarchy for the RefSeq annotated human reference genome which \n",
      "has the assembly accession GCF_000001405.30 and assembly name GRCh38.p4 looks \n",
      "like this:\n",
      "   /genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.30_GRCh38.p4\n",
      "\n",
      "Genome assemblies of interest can be identified using the NCBI Assembly resource\n",
      "(www.ncbi.nlm.nih.gov/assembly), or by using the assembly summary report files \n",
      "that are provided for both all genbank and all refseq assemblies:\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/ASSEMBLY_REPORTS/assembly_summary_genbank.txt\n",
      "or ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly_summary_genbank.txt\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/ASSEMBLY_REPORTS/assembly_summary_refseq.txt\n",
      "or ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/assembly_summary_refseq.txt\n",
      "\n",
      "Assembly summary report files containing information on assemblies for a \n",
      "particular taxonomic group or species are provided in the group and \n",
      "Genus_species directories under the \"genbank\" and \"refseq\" directory trees. e.g.\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt\n",
      "ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/Sulfolobus_islandicus/assembly_summary.txt\n",
      "\n",
      "Search the meta-data fields, or filter the files, to find assemblies of \n",
      "interest.\n",
      "\n",
      "\n",
      "===========================\n",
      "Data provided per assembly:\n",
      "===========================\n",
      "Sequence and other data files provided per assembly are named according to the \n",
      "rule:\n",
      "[assembly accession.version]_[assembly name]_[content type].[optional format]\n",
      "\n",
      "File formats and content:\n",
      "\n",
      "   *_assembly_report.txt file\n",
      "       Tab-delimited text file reporting the name, role and sequence \n",
      "       accession.version for objects in the assembly. The file header contains \n",
      "       meta-data for the assembly including: assembly name, assembly \n",
      "       accession.version, scientific name of the organism and its taxonomy ID, \n",
      "       assembly submitter, and sequence release date. The file header also \n",
      "       indicates whether the assembly version is \"latest\", \"replaced\", or \n",
      "       \"suppressed\".\n",
      "   *_assembly_stats.txt file\n",
      "       Tab-delimited text file reporting statistics for the assembly including: \n",
      "       total length, ungapped length, contig & scaffold counts, contig-N50, \n",
      "       scaffold-L50, scaffold-N50, scaffold-N75, and scaffold-N90\n",
      "   *_assembly_regions.txt\n",
      "       Provided for assemblies that include alternate or patch assembly units. \n",
      "       Tab-delimited text file reporting the location of genomic regions and \n",
      "       the alt/patch scaffolds placed within those regions.\n",
      "   *_assembly_structure directory\n",
      "       This directory will only be present if the assembly has internal \n",
      "       structure. When present, it will contain AGP files that define how \n",
      "       component sequences are organized into scaffolds and/or chromosomes. \n",
      "       Other files define how scaffolds and chromosomes are organized into \n",
      "       non-nuclear and other assembly-units, and how any alternate or patch \n",
      "       scaffolds are placed relative to the chromosomes. Refer to the README.txt\n",
      "       file in the assembly_structure directory for additional information.\n",
      "   *_feature_table.txt.gz\n",
      "       Tab-delimited text file reporting locations and attributes for a subset \n",
      "       of annotated features. Included feature types are: gene, CDS, RNA (all \n",
      "       types), operon, C/V/N/S_region, and V/D/J_segment. Replaces the .ptt & \n",
      "       .rnt format files that were provided in the old genomes FTP directories.\n",
      "       See the \"Description of files\" section below for details of the file \n",
      "       format.\n",
      "   *_genomic.fna.gz file\n",
      "       FASTA format of the genomic sequence(s) in the assembly. Repetitive \n",
      "       sequences in eukaryotes are masked to lower-case (see below).\n",
      "       The FASTA title is formatted as sequence accession.version plus \n",
      "       description. The genomic.fna.gz file includes all top-level sequences in\n",
      "       the assembly (chromosomes, plasmids, organelles, unlocalized scaffolds,\n",
      "       unplaced scaffolds, and any alternate loci or patch scaffolds). Scaffolds\n",
      "       that are part of the chromosomes are not included because they are\n",
      "       redundant with the chromosome sequences; sequences for these placed \n",
      "       scaffolds are provided under the assembly_structure directory.\n",
      "   *_genomic.gbff.gz file\n",
      "       GenBank flat file format of the genomic sequence(s) in the assembly. This\n",
      "       file includes both the genomic sequence and the CONTIG description (for \n",
      "       CON records), hence, it replaces both the .gbk & .gbs format files that \n",
      "       were provided in the old genomes FTP directories.\n",
      "   *_genomic.gff.gz file\n",
      "       Annotation of the genomic sequence(s) in Generic Feature Format Version 3\n",
      "       (GFF3). Sequence identifiers are provided as accession.version.\n",
      "       Additional information about NCBI's GFF files is available at \n",
      "       ftp://ftp.ncbi.nlm.nih.gov/genomes/README_GFF3.txt.\n",
      "   *_protein.faa.gz file\n",
      "       FASTA format of the accessioned protein products annotated on the genome \n",
      "       assembly\n",
      "       The FASTA title is formatted as sequence accession.version plus \n",
      "       description.\n",
      "   *_protein.gpff.gz file\n",
      "       GenPept format of the accessioned protein products annotated on the \n",
      "       genome assembly\n",
      "   *_rm.out.gz file\n",
      "       RepeatMasker output; \n",
      "       Provided for Eukaryotes \n",
      "   *_rm.run file\n",
      "       Documentation of the RepeatMasker version, parameters, and library; \n",
      "       Provided for Eukaryotes \n",
      "   *_rna.fna.gz file\n",
      "       FASTA format of accessioned RNA products annotated on the genome \n",
      "       assembly; Provided for RefSeq assemblies as relevant (Note, RNA and mRNA \n",
      "       products are not instantiated as a separate accessioned record in GenBank\n",
      "       but are provided for some RefSeq genomes, most notably the eukaryotes.)\n",
      "       The FASTA title is provided as sequence accession.version plus \n",
      "       description.\n",
      "   *_rna.gbff.gz file\n",
      "       GenBank flat file format of RNA products annotated on the genome \n",
      "       assembly; Provided for RefSeq assemblies as relevant\n",
      "   *_wgsmaster.gbff.gz\n",
      "       GenBank flat file format of the WGS master for the assembly (present only\n",
      "       if a WGS master record exists for the sequences in the assembly).\n",
      "   md5checksums.txt file\n",
      "       file checksums are provided for all data files in the directory\n",
      "\n",
      "\n",
      "=====================\n",
      "Description of files:\n",
      "=====================\n",
      "\n",
      "Masking of fasta sequences in genomic.fna.gz files\n",
      "--------------------------------------------------\n",
      "Repetitive sequences in eukaryotic genome assembly sequence files, as \n",
      "identified by WindowMasker (Morgulis A, Gertz EM, Schaffer AA, Agarwala R. \n",
      "2006. Bioinformatics 22:134-41), have been masked to lower-case.\n",
      "\n",
      "Alignment programs typically have parameters that control whether the program \n",
      "will ignore lower-case masking, treat it as soft-masking (i.e. only for finding \n",
      "initial matches) or treat it as hard-masking. By default NCBI BLAST will ignore \n",
      "lower-case masking but this can be changed by adding options to the blastn \n",
      "command-line.\n",
      "To have blastn treat lower-case masking in the query sequence as soft-masking \n",
      "add:\n",
      "     -lcase_masking\n",
      "To have blastn treat lower-case masking in the query sequence as hard-masking \n",
      "add:\n",
      "     -lcase_masking -soft_masking false\n",
      "\n",
      "Alternatively, commands such as the following can be used to generate either \n",
      "unmasked sequence or sequence masked with Ns.\n",
      "\n",
      "Example commands to remove lower-case masking:\n",
      "perl -pe '/^[^>]/ and $_=uc' genomic.fna > genomic.unmasked.fna\n",
      "  -or-\n",
      "awk '{if(/^[^>]/)$0=toupper($0);print $0}' genomic.fna > genomic.unmasked.fna\n",
      "\n",
      "Example commands to convert lower-case masking to masking with Ns (hard-masked):\n",
      "perl -pe '/^[^>]/ and $_=~ s/[a-z]/N/g' genomic.fna > genomic.N-masked.fna\n",
      "  -or-\n",
      "awk '{if(/^[^>]/)gsub(/[a-z]/,\"N\");print $0}' genomic.fna > genomic.N-masked.fna\n",
      "\n",
      "\n",
      "*_feature_table.txt.gz\n",
      "----------------------\n",
      "Tab-delimited text file reporting locations and attributes for a subset of \n",
      "annotated features. Included feature types are: gene, CDS, RNA (all types), \n",
      "operon, C/V/N/S_region, and V/D/J_segment. \n",
      "\n",
      "The file is tab delimited (including a #header) with the following columns:\n",
      "col 1: feature: INSDC feature type\n",
      "col 2: class: For ncRNA features, this is the ncRNA_class for the feature. For \n",
      "       gene features, this is the gene biotype computed based on the set of \n",
      "       child features for that gene. See the description of the gene_biotype \n",
      "       attribute in the GFF3 documentation for more details:\n",
      "       ftp://ftp.ncbi.nlm.nih.gov/genomes/README_GFF3.txt\n",
      "col 3: assembly: assembly accession.version\n",
      "col 4: assembly_unit: name of the assembly unit, such as \"Primary Assembly\", \n",
      "       \"ALT_REF_LOCI_1\", or \"non-nuclear\"\n",
      "col 5: seq_type: sequence type, computed from the \"Sequence-Role\" and \n",
      "       \"Assigned-Molecule-Location/Type\" in the *_assembly_report.txt file. The\n",
      "       value is computed as:\n",
      "       if an assembled-molecule, then reports the location/type value. e.g. \n",
      "       chromosome, mitochondrion, or plasmid\n",
      "       if an unlocalized-scaffold, then report \"unlocalized scaffold on <type>\".\n",
      "       e.g. unlocalized scaffold on chromosome\n",
      "       else the role, e.g. alternate scaffold, fix patch, or novel patch\n",
      "col 6: chromosome\n",
      "col 7: genomic_accession\n",
      "col 8: start: feature start coordinate (base-1). start is always less than end\n",
      "col 9: end: feature end coordinate (base-1)\n",
      "col10: strand\n",
      "col11: product_accession: accession.version of the product referenced by this \n",
      "       feature, if exists\n",
      "col12: non-redundant_refseq: for bacteria and archaea assemblies, the \n",
      "       non-redundant WP_ protein accession corresponding to the CDS feature. May\n",
      "       be the same as column 11, for RefSeq genomes annotated directly with WP_\n",
      "       RefSeq proteins, or may be different, for genomes annotated with \n",
      "       genome-specific protein accessions (e.g. NP_ or YP_ RefSeq proteins) that\n",
      "       reference a WP_ RefSeq accession.\n",
      "col13: related_accession: for eukaryotic RefSeq annotations, the RefSeq protein\n",
      "       accession corresponding to the transcript feature, or the RefSeq \n",
      "       transcript accession corresponding to the protein feature.\n",
      "col14: name: For genes, this is the gene description or full name. For RNA, CDS,\n",
      "       and some other features, this is the product name.\n",
      "col15: symbol: gene symbol\n",
      "col16: GeneID: NCBI GeneID, for those RefSeq genomes included in NCBI's Gene \n",
      "       resource\n",
      "col17: locus_tag\n",
      "col18: feature_interval_length: sum of the lengths of all intervals for the \n",
      "       feature (i.e. the length without introns for a joined feature)\n",
      "col19: product_length: length of the product corresponding to the \n",
      "       accession.version in column 11. Protein product lengths are in amino acid\n",
      "       units, and do not include the stop codon which is included in column 18.\n",
      "       Additionally, product_length may differ from feature_interval_length if \n",
      "       the product contains sequence differences vs. the genome, as found for \n",
      "       some RefSeq transcript and protein products based on mRNA sequences and \n",
      "       also for INSDC proteins that are submitted to correct genome \n",
      "       discrepancies.\n",
      "col20: attributes: semi-colon delimited list of a controlled set of qualifiers.\n",
      "       The list currently includes:\n",
      "       partial, pseudo, pseudogene, ribosomal_slippage, trans_splicing, \n",
      "       anticodon=NNN (for tRNAs), old_locus_tag=XXX \n",
      "\n",
      "________________________________________________________________________________\n",
      "National Center for Biotechnology Information (NCBI)\n",
      "National Library of Medicine\n",
      "National Institutes of Health\n",
      "8600 Rockville Pike\n",
      "Bethesda, MD 20894, USA\n",
      "tel: (301) 496-2475\n",
      "fax: (301) 480-9241\n",
      "e-mail: info@ncbi.nlm.nih.gov\n",
      "________________________________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "!cat README.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
